src/protocol.h

```diff
@@ -471,12 +475,14 @@ enum GetDataMsg : uint32_t {
     UNDEFINED = 0,
     MSG_TX = 1,
     MSG_BLOCK = 2,
-    MSG_WTX = 5,                                      //!< Defined in BIP 339
     // The following can only occur in getdata. Invs always use TX/WTX or BLOCK.
     MSG_FILTERED_BLOCK = 3,                           //!< Defined in BIP37
     MSG_CMPCT_BLOCK = 4,                              //!< Defined in BIP152
+    MSG_DANDELION_TX = 5,
+    MSG_WTX = 6,                                      //!< Defined in BIP 339
```

Comment:
What was the reason `MSG_WTX` was changed to `0x06`?
Was that required?

Let me guess: 
`MSG_DANDELION_TX` was set to `0x05` before. If we decide to stick to the BIP standard, the new node version `8.22` won't be backwards compatible when someone broadcasts a `MSG_WTX`.

If that is the case, I think we are screwed and we should add a comment explaining that in DigiByte, the MSG_WTX was changed to `0x06` because Dandelion got introduced in between

Reply:
Yeah, MSG_DANDELION_TX = 5 was 5 in 7.17.2/3. However MSG_WTX was effectively 1 if I am reading this correctly. 

```
enum GetDataMsg
{
    UNDEFINED = 0,
    MSG_TX = 1,
    MSG_BLOCK = 2,
    // The following can only occur in getdata. Invs always use TX or BLOCK.
    MSG_FILTERED_BLOCK = 3,  //!< Defined in BIP37
    MSG_CMPCT_BLOCK = 4,     //!< Defined in BIP152
    MSG_DANDELION_TX = 5,    //!< Dandelion
    MSG_WITNESS_BLOCK = MSG_BLOCK | MSG_WITNESS_FLAG, //!< Defined in BIP144
    MSG_WITNESS_TX = MSG_TX | MSG_WITNESS_FLAG,       //!< Defined in BIP144
    MSG_FILTERED_WITNESS_BLOCK = MSG_FILTERED_BLOCK | MSG_WITNESS_FLAG,
    MSG_DANDELION_WITNESS_TX = MSG_DANDELION_TX | MSG_WITNESS_FLAG,
};
```

https://github.com/DigiByte-Core/digibyte/blob/develop/src/protocol.h#L384

Reply:
`MSG_WTX` was introduced in Bitcoin upstream some time ago. According to the BIP standard it has to be `5`.
So if we change it to `6`, we have to at least put a comment there. That's what I am saying!

Better would be to change `MSG_DANDELION_TX` to something different like 10. That however won't be backwards compatible to our old client.

Reply:
@SmartArray yep, nailed it in one. The enumeration set expects MSG_DANDELION_TX to be 5 (wtxid relay didnt exist back then), this value actually gets encoded into an inventory item.
Having it as it is now allows compatibility with existing clients. As long as we remember to keep MSG_WTX at 6 for future clients, there won't be any issues.